mods/ra/maps/exodus/exodus.lua

```diff
@@ -0,0 +1,394 @@
+		local circling = Utils.Where(sovietYaks, function(yak) return yak.IsOnHoldingPattern end)
+		if #circling > 0 then
+			Utils.Do(circling, YakAttack)
+		end
+
+		local grounded = Utils.Where(sovietYaks, function(yak)
+			return yak.CenterPosition.Z < 1 and yak.AmmoCount() == yak.MaximumAmmoCount()
+		end)
+		if #grounded > 0 then
+			Utils.Do(grounded, YakAttack)
+		end
+	end
+end
+
+UnitsEvacuated = 0
+EvacuateAlliedUnit = function(unit)
+	if (unit.Owner == allies1 or unit.Owner == allies2) and unit.IsMobile then
+		unit.Stop()
+		unit.Owner = allies
+
+		if unit.Type == 'mgg' then
+			Utils.Do(humans, function(player)
+				if player then
+					player.MarkCompletedObjective(evacuateMgg)
+				end
+			end)
+		end
+
+		UnitsEvacuated = UnitsEvacuated + 1
+		if unit.HasProperty("HasPassengers") then
+			UnitsEvacuated = UnitsEvacuated + unit.PassengerCount
+		end
+
+		local exitCell = Map.ClosestEdgeCell(unit.Location)
+		Trigger.OnIdle(unit, function()
+			unit.ScriptedMove(exitCell)
+		end)
+
+		local exit = Map.CenterOfCell(exitCell)
+		Trigger.OnEnteredProximityTrigger(exit, WDist.FromCells(1), function(a)
+			a.Destroy()
+		end)
+
+		UserInterface.SetMissionText(UnitsEvacuated .. "/" .. unitsEvacuatedThreshold .. " units evacuated.")
+
+		if UnitsEvacuated >= unitsEvacuatedThreshold then
+			Utils.Do(humans, function(player)
+				if player then
+					player.MarkCompletedObjective(evacuateUnits)
+				end
+			end)
+		end
+	end
+end
+
+Tick = function()
+	if DateTime.GameTime % 100 == 0 then
+		ManageSovietUnits()
+		ManageSovietAircraft()
+
+		Utils.Do(humans, function(player)
+			if player and player.HasNoRequiredUnits() then
+				soviets.MarkCompletedObjective(sovietObjective)
+			end
+		end)
+	end
+end
+
+WorldLoaded = function()
+	-- NPC
+	neutral = Player.GetPlayer("Neutral")
+	allies = Player.GetPlayer("Allies")
+	soviets = Player.GetPlayer("Soviets")
+
+	-- Player controlled
+	allies1 = Player.GetPlayer("Allies1")
+	allies2 = Player.GetPlayer("Allies2")
+
+	humans = { allies1, allies2 }
+	Utils.Do(humans, function(player)
+		if player and player.IsLocalPlayer then
+			Trigger.OnObjectiveAdded(player, function(p, id)
+				Media.DisplayMessage(p.GetObjectiveDescription(id), "New " .. string.lower(p.GetObjectiveType(id)) .. " objective")
+			end)
+
+			Trigger.OnObjectiveCompleted(player, function(p, id)
+				Media.DisplayMessage(p.GetObjectiveDescription(id), "Objective completed")
+			end)
+
+			Trigger.OnObjectiveFailed(player, function(p, id)
+				Media.DisplayMessage(p.GetObjectiveDescription(id), "Objective failed")
+			end)
+
+			Trigger.OnPlayerWon(player, function()
+				Media.PlaySpeechNotification(player, "MissionAccomplished")
+			end)
+
+			Trigger.OnPlayerLost(player, function()
+				Media.PlaySpeechNotification(player, "MissionFailed")
+			end)
+		end
+	end)
+
+	unitsEvacuatedThreshold = UnitsEvacuatedThreshold[Difficulty]
+	UserInterface.SetMissionText(UnitsEvacuated .. "/" .. unitsEvacuatedThreshold .. " units evacuated.")
```

Comment:
I think it would look better if we use the player color from the allied campaign here. Just visual and just me though.